### PR TITLE
Options for count

### PR DESCRIPTION
**Pull request by intech**
Options do not work! You can not work properly without [_routing](https://www.elastic.co/blog/customizing-your-document-routing).

### CODE REVIEW

**Comment by ruflin** [comment]
@intech Thanks for the addition. Could you update the CHANGELOG.md file and add a small tests that would not have worked before and passes now?

**Comment by intech** [comment]
@ruflin ok! ;)

**Comment by intech** [comment]
@ruflin Done!

**Comment by ruflin** [comment]
@intech Seems like your change breaks an other test: https://travis-ci.org/ruflin/Elastica/jobs/102967963#L2086

**Comment by intech** [comment]
@ruflin i'm sorry, it's my fault!

**Comment by intech** [comment]
hm...

**Comment by intech** [comment]
hm....

**Comment by intech** [comment]
@ruflin help me)
https://github.com/intech/Elastica/blob/patch-2/test/lib/Elastica/Test/SearchTest.php#L388 
Where did r1, r2? The documents do not have them.
These tests are working correctly and my returns all the documents.
My commits precisely fix the problem in my project it works.

**Comment by intech** [comment]
Oh!!! Default limit 10! One minutes...

**Comment by intech** [comment]
Paranormal with routing. On the project, everything works fine, and the testing is not considered routing and parent
Try again tomorrow with a fresh mind to see again.

**Comment by intech** [comment]
I did commit their failed attempts. Maybe someone will see what is my fault?

File: test/lib/Elastica/Test/SearchTest.php [hunk 1]
```diff
@@ -450,19 +451,12 @@ public function testCountRequest()
...
-            new Document(6,  array('id' => 1, 'email' => 'test@test.com', 'username' => 'marley')),
-            new Document(7,  array('id' => 1, 'email' => 'test@test.com', 'username' => 'marley')),
-            new Document(8,  array('id' => 1, 'email' => 'test@test.com', 'username' => 'marley')),
-            new Document(9,  array('id' => 1, 'email' => 'test@test.com', 'username' => 'marley')),
-            new Document(10, array('id' => 1, 'email' => 'test@test.com', 'username' => 'marley')),
-            new Document(11, array('id' => 1, 'email' => 'test@test.com', 'username' => 'marley')),
-        ));
+
+        $docs = array();
+        for ($i = 0; $i < 11; ++$i) {
```
Comment (damienalexandre) [inline test/lib/Elastica/Test/SearchTest.php]: Should start `$i` at `1` to be the same (and avoid document with id="0").

**Comment by ruflin** [comment]
This PR seems to overlap with: https://github.com/ruflin/Elastica/pull/1040

**Comment by ruflin** [comment]
@intech @vinusebastian Any updates here? I would be nice to get at least one of the two PR's into Elastica.

**Comment by ruflin** [comment]
I opened a new PR here based on this one as I would like to get these fixes in: https://github.com/ruflin/Elastica/pull/1164

**Comment by ruflin** [comment]
I'm closing this one in favor of https://github.com/ruflin/Elastica/pull/1164 @intech Thanks a lot for your work. If you have some ideas why https://github.com/ruflin/Elastica/pull/1164 is failing, let me know.

**Pull request closed without merging**